File: Ejercicios Propuestos 2/Ejercicio 21/Ejercicio 21.jl
println("Ingrese un numero entero:")
num = parse(Int, readline())

invertido = 0
# si se da el caso de que el número es negativo, extraemos el signo
# para trabajar con su valor absoluto y no tener errores en el módulo.
es_negativo = num < 0
if es_negativo
    global num
    num = num * (-1)
end

while num > 0
    global num, invertido
    
    resto = num % 10
    invertido = (invertido * 10) + resto
    num = div(num, 10)
end

# Restauramos el signo negativo si correspondía
if es_negativo
    invertido = invertido * (-1)
end

println("El numero invertido es: ", invertido)

File: Ejercicios Propuestos 2/Ejercicio 21/ejercicio21.cpp
#include <iostream>
using namespace std;

int main() {
    int num, invertido = 0, resto;
    cout << "Ingrese un numero entero: ";
    cin >> num;

    bool es_negativo = num < 0;
    if (es_negativo) {
        num = -num;
    }

    while (num > 0) {
        resto = num % 10;
        invertido = (invertido * 10) + resto;
        num /= 10;
    }

    if (es_negativo) {
        invertido = -invertido;
    }

    cout << "El numero invertido es: " << invertido << endl;
    return 0;
}